docs/design/run.md

```diff
@@ -34,11 +33,13 @@ the git-hook model, there was consensus about not repeating their mistakes.
 
 For `jj run` there is prior art in Mercurial, git branchless and Google's 
 internal Mercurial. Currently git-branchless `git test` and `hg fix` implement
-some kind of command runner. While the Google internal `hg run` works in 
+some kind of command runner. The Google internal `hg run` works in 
 conjunction with CitC (Clients in the Cloud) which allows it to lazily apply
-the current command to any affected file. The base Jujutsu backend does not
-have a fancy virtual filesystem supporting it, so we can't apply this 
-optimization. 
+the current command to any affected file. Currently no Jujutsu backend
+(Git, Native) has a fancy virtual filesystem supporting it, so we 
+can't apply this optimization. We could do the same once we have an 
```

Comment:
FYI (no change needed): I don't think a VFS would need to be tied to a commit backend. The working-copy backend we're going to add for Google will be specific to CitC but one could imagine a working-copy backend that's independent of commit backend.

Reply:
Technically a VFS is not required to be bound to a commit backend, but it is the common option for traditional VCS. 
Examples: [GVFS/VFSForGit] and [EdenFS].

Only Jujutsu has a new notion of a working-copy which allows different backends, so I'd say the reason still stands on its own. 

> The working-copy backend we're going to add for Google will be specific to CitC but one could imagine a working-copy backend that's independent of commit backend.

Yes that seems reasonable to me, a wc-copy independent of the commit backend seems a good idea for some use-cases.  

[EdenFS]: https://github.com/facebook/sapling/blob/0fa83e83ac781f2bc64b26192d365280c11d4dce/eden/fs/service/eden.thrift#L1573
[GVFS/VFSForGit]: https://github.com/microsoft/VFSForGit